nvme.c

```diff
@@ -5384,6 +5384,8 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
 	if (err < 0) {
 		nvme_show_error("Invalid output format");
 		goto free_tree;
+	} else {
+		err = 0 ;
```

Comment:
How about this instead (moving `err = flags` into the `if` statement)?
```c
flags = validate_output_format(output_format_val);
if (flags < 0) {
	nvme_show_error("Invalid output format");
	err = flags;
	goto free_tree;
}
```

Reply:
That's a good idea. 
However, @igaw changed the validate_output_format function #2142 
thank you

Reply:
@calebsander initially `flags` was just defined as `int` and I changed it to an enum but didn't update the parse function at this point. I don't really like to mix the different types, thus it was about time to also address this by changing the parse function so that return is just an error code and the `flags` variable is passed in as argument.